test/cli/package-special-allowed-dsls/foo/__package.rb

```diff
@@ -3,6 +3,4 @@
 class Project::Foo < PackageSpec
   restrict_to_service Project::Foo
   restrict_to_service Project::Bar
-
-  autoloader_compatibility 'strict'
```

Comment:
Do you think it's worth keeping at least one mention of `autoloader_compatibility 'strict'` in the test suite to assert that we produce an error on it?